src/test/java/com/inventorsoft/junit/service/BookServiceTest.java

```diff
@@ -0,0 +1,222 @@
+    static final LocalDate AUTHOR_BIRTHDAY = LocalDate.of(1947, 9, 21);
+
+    static Author author = new Author();
+    static Book book = new Book();
+    static BookResponse bookResponse = new BookResponse();
+    static AuthorResponse authorResponse = new AuthorResponse();
+    static UpdateBookRequest updateBookRequest = new UpdateBookRequest();
+
+    @BeforeEach
+    void init() {
+        author.setBirthday(AUTHOR_BIRTHDAY);
+        author.setName(AUTHOR_NAME);
+        author.setId(AUTHOR_ID);
+
+        book.setAuthor(author);
+        book.setDescription(DESCRIPTION);
+        book.setReleaseDate(RELEASE_DATE);
+        book.setId(ID);
+        book.setTitle(TITLE);
+
+        authorResponse.setBirthDate(AUTHOR_BIRTHDAY);
+        authorResponse.setName(AUTHOR_NAME);
+        authorResponse.setId(AUTHOR_ID);
+
+        bookResponse.setId(ID);
+        bookResponse.setTitle(TITLE);
+        bookResponse.setReleaseDate(RELEASE_DATE);
+        bookResponse.setDescription(DESCRIPTION);
+        bookResponse.setAuthor(authorResponse);
+    }
+
+    @Test
+    void getById() {
+        when(bookRepository.findById(ID)).thenReturn(Optional.of(book));
+        when(bookMapper.mapEntityToResponse(book)).thenReturn(bookResponse);
+        BookResponse br = bookService.getById(ID);
+
+        assertEquals(bookResponse, br);
+    }
+
+    @Test
+    void getByIdMustThrowExceptionIfIdIsMissing() {
+        assertThrows(RuntimeException.class, () -> bookService.getById(null));
+    }
+
+    @Test
+    void getAll() {
+        List<Book> bookList = List.of(book);
+        List<BookResponse> bookResponseList = List.of(bookResponse);
+
+        when(bookRepository.findAll()).thenReturn(bookList);
+        when(bookMapper.mapEntitiesToResponses(bookList)).thenReturn(bookResponseList);
+
+        List<BookResponse> actualList = bookService.getAll();
+
+        for (int i = 0; i < bookList.size() && i < actualList.size(); i++) {
+            assertEquals(bookResponseList.get(i).getTitle(), actualList.get(i).getTitle());
+            assertEquals(bookResponseList.get(i).getDescription(), actualList.get(i).getDescription());
+            assertEquals(bookResponseList.get(i).getReleaseDate(), actualList.get(i).getReleaseDate());
+            assertEquals(bookResponseList.get(i).getAuthor(), actualList.get(i).getAuthor());
+        }
+    }
+
+    @Test
+    void create() {
+        CreateBookRequest createBookRequest = new CreateBookRequest();
+        createBookRequest.setAuthorId(AUTHOR_ID);
+        createBookRequest.setTitle(TITLE);
+        createBookRequest.setDescription(DESCRIPTION);
+        createBookRequest.setReleaseDate(RELEASE_DATE);
+
+        when(bookMapper.mapCreateRequestToEntity(createBookRequest)).thenReturn(book);
+        when(bookRepository.save(book)).thenReturn(book);
+
+        Long result = bookService.create(createBookRequest);
+
+        assertEquals(ID, result);
+        verify(bookRepository, times(1)).save(book);
```

Comment:
just for you to know, that will do the same
```suggestion
        verify(bookRepository).save(book);
```